pkg/jamsession/jamsession.go

```diff
@@ -72,11 +73,15 @@ type JamSession interface {
 	// SetPlayerState TODO
 	SetPlayerState(*spotify.PlayerState)
 	// GetDeviceID TODO
-	GetDeviceID() spotify.ID
+	GetDevice() spotify.PlayerDevice
 	// SetDeviceID TODO
-	SetDeviceID(spotify.ID)
-	// SetDevice TODO
-	SetDevice(id string) error
+	SetDevice(string) error
+	// SocketJamUpdate updates the JamSession through the Websocket connection
+	SocketJamUpdate()
+	// SocketQueueUpdate updates the Queue through the Websocket connection
+	SocketQueueUpdate()
+	// SocketPlaybackUpdate updates the Playback through the Websocket connection
+	SocketPlaybackUpdate()
```

Comment:
We should think about what methods we actually need in the interface and what methods are specific to the Spotify implementations. No changes required here for now though.

Reply:
Yes, the interface currently is a mess. But I think it's still better to repeat the same code multiple times to send the same websocket message on different occasions

Reply:
I agree on both statements.